### DISCUSSION TITLE
jdl调用paddleOCR

### DISCUSSION THREAD

**Discussion by CrazyShado** [hiroi-sora/Umi-OCR#520] [General]
您好，我当前在通过java方式实现ocr的识别。使用的是DJL(Deep Java Library)调用的paddle ocrv4的模型，整体跑下来图片的识别时间比较久1.2s左右，同一张图片用uniOCR识别下来再0.5s，所以想咨询下，您在识别开始的时候，都有做哪些前置处理，比如图片尺寸的压缩，质量的调整，灰度处理等。感谢

**Comment by hiroi-sora** [comment]
我不太清楚 Paddle Java 接口底层使用的是什么推理库。据我的实测，Umi当前使用的 Paddle C++ MKL 是最快的推理库（当然，CPU和内存占用也是最高的）。别的，比如基于onnx的推理库，比如 FastDeploy 的后端，比如 Python 包附带的推理库，都比 MKL 有一定差距。你的识别耗时差异，可能是推理库区别造成的。

至于预处理，只做了尺寸压缩这一项，如 `软件标签页→设置→限制图片边长` 中所示。

Paddle库的训练样本是含三通道图片的，所以转灰度也没太大必要，模型本身适应彩色图像。

**Reply by CrazyShado** [reply]
好呢，非常感谢。
